hazelcast-code-generator/src/main/java/com/hazelcast/client/protocol/generator/CodecModel.java

```diff
@@ -153,9 +154,9 @@ public CodecModel(TypeElement parent, ExecutableElement methodElement, Executabl
         this.lang = lang;
 
         name = methodElement.getSimpleName().toString();
-        short requestId = methodElement.getAnnotation(Request.class).id();
+        requestId = methodElement.getAnnotation(Request.class).id();
         short masterId = parent.getAnnotation(GenerateCodec.class).id();
-        id = CodeGenerationUtils.mergeIds(masterId, requestId);
+        id = CodeGenerationUtils.addHexPrefix(CodeGenerationUtils.mergeIds(masterId, requestId));
```

Comment:
I subtracted the method from mergeId un purpose because I needed the int without hex prefix somewhere else. I dont see any value creating a new method like `mergeIdsWithHexPrefix`. 
This way two concerns are in separate methods.